p2p/dial.go

```diff
@@ -103,26 +104,30 @@ func newDialState(static []*discover.Node, ntab discoverTable, maxdyn int) *dial
 	s := &dialstate{
 		maxDynDials: maxdyn,
 		ntab:        ntab,
-		static:      make(map[discover.NodeID]*discover.Node),
+		static:      make(map[discover.NodeID]*dialTask),
 		dialing:     make(map[discover.NodeID]connFlag),
 		randomNodes: make([]*discover.Node, maxdyn/2),
 		hist:        new(dialHistory),
 	}
 	for _, n := range static {
-		s.static[n.ID] = n
+		s.addStatic(n)
 	}
 	return s
 }
 
 func (s *dialstate) addStatic(n *discover.Node) {
-	s.static[n.ID] = n
+	// This overwites the task instead of updating an existing
+	// entry, giving users the opportunity to force a resolve operation.
+	s.static[n.ID] = &dialTask{flags: staticDialedConn, dest: n}
 }
 
 func (s *dialstate) newTasks(nRunning int, peers map[discover.NodeID]*Peer, now time.Time) []task {
 	var newtasks []task
+	isDialing := func(id discover.NodeID) bool {
+		return s.dialing[id] != 0 || peers[id] != nil || s.hist.contains(id)
```

Comment:
Hmm, this `s.dialing[id] != 0` is not *that* obvious why it works, given that it's a `connFlag`. Imho it would be a bit more future proof to retain the old `_, dialing := s.dialing[id]` implementation. That way if we assign some meaning to an unset connFlag, it will still work ok.